src/charm.py

```diff
@@ -120,15 +142,33 @@ def _login_ui_layer(self) -> Layer:
         return Layer(pebble_layer)
 
     def _get_kratos_endpoint_info(self) -> Optional[str]:
-        kratos_public_url = ""
         if self.model.relations[self._kratos_relation_name]:
             try:
                 kratos_endpoints = self.kratos_endpoints.get_kratos_endpoints()
                 kratos_public_url = kratos_endpoints["public_endpoint"]
+                return kratos_public_url
             except KratosEndpointsRelationDataMissingError:
                 logger.info("No kratos-endpoint-info relation data found")
+                return ""
+        return ""
+
+    def _update_login_ui_endpoint_relation_data(self, event: RelationEvent) -> None:
+        endpoint = self.ingress.url if self.ingress.is_ready() else ""
 
-        return kratos_public_url
+        logger.info(f"Sending login ui endpoint info: endpoint - {endpoint[0]}")
+
+        self.endpoints_provider.send_endpoints_relation_data(endpoint[0])
+
+    def _get_hydra_endpoint_info(self) -> Optional[str]:
+        if self.model.relations[self._hydra_relation_name]:
+            try:
+                hydra_endpoints = self.hydra_endpoints.get_hydra_endpoints()
+                hydra_url = hydra_endpoints["public_endpoint"]
+                return hydra_url
+            except HydraEndpointsRelationDataMissingError:
+                logger.info("No hydra endpoint-info relation data found")
+                return ""
+        return ""
```

Comment:
I'm being nit-picky, but I think this would be cleaner if you didn't have so many `return` statements. Something like this:

```suggestion
    def _get_hydra_endpoint_info(self) -> Optional[str]:
        hydra_url = ""
        try:
            hydra_endpoints = self.hydra_endpoints.get_hydra_endpoints()
            hydra_url = hydra_endpoints["public_endpoint"]
        except HydraEndpointsRelationDataMissingError:
            logger.info("No hydra endpoint-info relation data found")
        return hydra_url
```

Reply:
gruyaume suggestion was to return `None` though, in that case you would have only 1 return statement.

I think that you must either return `""` here, in which case you would need this approach, or you must return None and handle that case in `_login_ui_layer` with something like:
```python
"HYDRA_ADMIN_URL": self._get_hydra_endpoint_info() or None,
```

Reply:
Sure, I may have confused this repo with another. I'm working simultaneously on 3 prs and the functions I wrote are basically the same. I tried to update all of them to have consistency, but because of different requirements (login_ui explicitly needs strings, while the other operators need None values if there's no value) they went out of sync. I'll submit your proposal here with the minor change of handling endpoints relation missing error.